src/bit_vec.rs

```diff
@@ -34,23 +30,18 @@ impl<O: BitOrder, T: BitStore> Encode for BitSlice<T, O>
 		);
 		Compact(bits as u32).encode_to(dest);
 
-		for element in self.domain() {
+		// Iterate over chunks
+		for chunk in self.chunks(core::mem::size_of::<T>() * 8) {
+			let mut element = T::ZERO;
+			element.view_bits_mut::<O>()[..chunk.len()].copy_from_bitslice(chunk);
```

Comment:
Nit: can `[..chunk.len()]` just be `[..]`?

Reply:
> If self.len() is not an even multiple of chunk_size, then the last chunk yielded will be shorter.

Reply:
aaah, I did read that, but I was misreading the code; makes sense now , thanks :)